src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,132 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        int i = 0;
+        int j = 0;
+        int x = 0;
+        int y = 0;
+
+        x = toX - robot.getX();
+        y = toY - robot.getY();
+
+        if (x > 0 && robot.getDirection() == Direction.UP) {
```

Comment:
let's simplify this algorithm
example for getting direction on x-axis:
```
if (toX > robot.getX()) {
            while (robot.getDirection() != Direction.RIGHT) {
                robot.turnRight();
            }
        }  else {
            while (robot.getDirection() != Direction.LEFT) {
                robot.turnRight();
            }
        }
```

Reply:
done.